qiime/plugin/testing.py

```diff
@@ -84,3 +90,33 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transform_format(self, source_format, target, **kwargs):
+        # Guard any non-QIIME2 Format sources from being tested
+        self.assertTrue(issubclass(source_format, FormatBase))
+
+        transformer = self.get_transformer(source_format, target)
+
+        source_path = None
+        if 'filename' in kwargs:
+            source_path = kwargs['filename']
+        elif 'filenames' in kwargs:
+            source_path = self.temp_dir.name
+            for filename in kwargs['filenames']:
+                filepath = self.get_data_path(filename)
+                shutil.copy(filepath, source_path)
+        else:
+            self.fail("The helper method `transform_format` requires either "
```

Comment:
I'm not sure if failing is what we want to do here, since this is a developer error, not a test failure. With the signature changes requested above you could do something like:

```python
if filename is not None and filenames is not None:
    raise ValueError("Cannot use both `filename` and `filenames` at the same time.")
```